frontend/src/contexts/broadcast-channel.context.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2025 Hexastack. All rights reserved.
+ *
+ * Licensed under the GNU Affero General Public License v3.0 (AGPLv3) with the following additional terms:
+ * 1. The name "Hexabot" is a trademark of Hexastack. You may not use this name in derivative works without express written permission.
+ * 2. All derivative works must include clear attribution to the original creator and software, Hexastack and Hexabot, in a prominent location (e.g., in the software's "About" section, documentation, and README file).
+ */
+
+import {
+  createContext,
+  FC,
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+} from "react";
+
+import { useTabUuid } from "@/hooks/useTabUuid";
+
+export type BroadcastChannelData = {
+  tabId: string;
+  data: string | number | boolean | Record<string, unknown> | undefined | null;
```

Comment:
```suggestion
  payload: BroadcastChannelPayload
```
Given that  type BroadcastChannelPayload = {
    event: string;
    data?: string | number | boolean | Record<string, unknown> | undefined | null;
    }